### PR TITLE
[codex] Handle settled Mux embed loader state

### PR DESCRIPTION
**Pull request by nathanjohnpayne**
Authoring-Agent: codex

## Summary

- Track the injected `mux-embed` script's load/error state with `data-mux-embed-status`.
- Let later Mux hero initializers proceed if an existing script has already settled, avoiding a stuck Promise before `<mux-background-video>` registration.

## Root Cause

CodeRabbit caught a valid edge case after PR #264 merged: if more than one Mux hero component initializes and one finds an existing `script[data-mux-embed]` after its load/error event has already fired, the listener-only path can wait forever.

## Validation

- `npm run test`
- `npm run test:e2e`

### CODE REVIEW

**Review by nathanpayne-codex — APPROVED** [review]
Reviewed as nathanpayne-codex.

No blockers found. This follow-up is narrowly scoped to the CodeRabbit edge case from #264: existing injected `mux-embed` scripts now carry a settled status so later initializers do not hang waiting for events that already fired. Local validation passed with `npm run test` and `npm run test:e2e`.

File: src/components/ProjectMuxPlayer.astro [hunk 1]
```diff
@@ -63,6 +63,13 @@ const posterSrc = playbackId
 
     const existingScript = document.querySelector('script[data-mux-embed]');
     if (existingScript) {
+      if (
+        existingScript.dataset.muxEmbedStatus === 'loaded' ||
+        existingScript.dataset.muxEmbedStatus === 'error'
+      ) {
+        return Promise.resolve();
+      }
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply src/components/ProjectMuxPlayer.astro:71]: [mergepath-resolve: deferred-to-followup] Confirmed still applicable. The settled-branch logic in `loadMuxEmbed()` (src/components/ProjectMuxPlayer.astro) is unchanged from what shipped in this PR, and the only test added since (commit 726c9e008f / PR #347, explicitly labeled as a #265 regression test) is a static string-match against the built module source (`expect(muxScript).toMatch(/muxEmbedStatus===['"]loaded['"]/)` etc.) -- it does not pre-seed a `script[data-mux-embed]` element, does not exercise the branch, and does not assert init resolves immediately with no new script appended, as this finding requested. Tracked in https://github.com/nathanjohnpayne/nathanpaynedotcom/issues/565.
